components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/MessageProcessorResource.java

```diff
@@ -0,0 +1,252 @@
+     **/
+    private JSONObject getMessageProcessorAsJson(MessageProcessor messageProcessor) {
+        JSONObject jsonBody = new JSONObject();
+        jsonBody.put(Constants.NAME, messageProcessor.getName());
+        jsonBody.put(TYPE_PROPERTY, getMessageProcessorType(messageProcessor));
+        jsonBody.put(CONTAINER_PROPERTY, messageProcessor.getArtifactContainerName());
+        jsonBody.put(FILE_NAME_PROPERTY, messageProcessor.getFileName());
+        jsonBody.put(PARAMETER_PROPERTY, messageProcessor.getParameters());
+        jsonBody.put(MESSAGE_STORE_PROPERTY, messageProcessor.getMessageStoreName());
+        jsonBody.put(STATUS_PROPERTY, getProcessorState(messageProcessor.isDeactivated()));
+
+        return jsonBody;
+    }
+
+    /**
+     * Returns the type of the message processor
+     * @param messageProcessor
+     * @return String type of the message processor
+     * */
+    private String getMessageProcessorType(MessageProcessor messageProcessor) {
+        if (messageProcessor instanceof ScheduledMessageForwardingProcessor) {
+            return SCHEDULED_MESSAGE_FORWARDING_TYPE;
+        } else if (messageProcessor instanceof FailoverScheduledMessageForwardingProcessor) {
+            return FAILOVER_SCHEDULED_MESSAGE_FORWARDING_TYPE;
+        } else if (messageProcessor instanceof SamplingProcessor) {
+            return SAMPLING_PROCESSOR_TYPE;
+        } else {
+            return CUSTOM_PROCESSOR_TYPE;
+        }
+    }
+
+    /**
+     * Changes the processor state to the specified state
+     * @param messageContext Axis2message context
+     * @param jsonObject message payload
+     * */
+    private void changeProcessorStatus(MessageContext messageContext, JsonObject jsonObject) {
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        SynapseConfiguration synapseConfiguration = messageContext.getConfiguration();
+        String processorName = jsonObject.get(PROCESSOR_NAME_PARAM).getAsString();
+        String status = jsonObject.get(STATUS_PROPERTY).getAsString();
+
+        MessageProcessor messageProcessor = synapseConfiguration.getMessageProcessors().get(processorName);
+        if (Objects.nonNull(messageProcessor)) {
+            JSONObject jsonResponse = new JSONObject();
+            if (PROCESSOR_INACTIVE_STATUS.equalsIgnoreCase(status)) {
+                messageProcessor.deactivate();
+                jsonResponse.put("Message", processorName + " : is deactivated");
+                Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
+                return;
+            }
+            if (PROCESSOR_ACTIVE_STATUS.equalsIgnoreCase(status)) {
+                messageProcessor.activate();
+                jsonResponse.put("Message", processorName + " : is activated");
+                Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
+                return;
+            }
```

Comment:
Shall we have an if-else ladder so that we can verify whether the provided input is correct? Additionally, shall we also refactor the code here to avoid duplicates?

Reply:
done